File: src/features/typing/data/java/lru_cache.java
import java.util.Iterator;
import java.util.LinkedHashMap;
import java.util.Map;

class LRUCache {
    private final int capacity;
    private final Map<Integer, Integer> cache = new LinkedHashMap<>(16, 0.75f, true);

    public LRUCache(int capacity) {
        this.capacity = capacity;
    }

    public int get(int key) {
        Integer value = cache.get(key);
        return value == null ? -1 : value;
    }

    public void put(int key, int value) {
        cache.put(key, value);
        if (cache.size() > capacity) {
            Iterator<Integer> keys = cache.keySet().iterator();
            cache.remove(keys.next());
        }
    }
}

File: src/features/typing/data/dart/lru_cache.dart
class LRUCache {
  final int _capacity;
  final Map<int, int> _cache = <int, int>{};

  LRUCache(int capacity) : _capacity = capacity;

  int get(int key) {
    if (!_cache.containsKey(key)) {
      return -1;
    }
    final value = _cache[key]!;
    _cache.remove(key);
    _cache[key] = value;
    return value;
  }

  void put(int key, int value) {
    _cache.remove(key);
    _cache[key] = value;
    if (_cache.length > _capacity) {
      _cache.remove(_cache.keys.first);
    }
  }
}
